packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -195,8 +206,18 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       endpoint.toBuilder(),
     );
     unawaited(
-      Amplify.asyncConfig.then((_) {
-        _endpointClient.updateEndpoint();
+      Amplify.asyncConfig.then((_) async {
+        try {
+          await _endpointClient.updateEndpoint();
+          _hubEventController.add(
+            AnalyticsHubEvent.endpointConfigured(fixedEndpointId),
+          );
+        } on Exception catch (e) {
+          throw AnalyticsException(
```

Comment:
This will result in an unhandled exception - there is no way for the user to catch this or see it in their code.

There are two solutions to this issue.

1. Instead of waiting for `Amplify.asyncConfig` we can add category-specific waiters, so that dependencies like this are easier to control, e.g. `Amplify.Auth.asyncConfig`
2. Instead of throwing the error, store it locally, then throw it whenever the first `Amplify.Analytics.*` method is called so that it is available to the customer.

Reply:
Got it thanks. 

I missed that you can't catch exceptions in unawaited futures. 

For 1) I feel that would add coupling between the categories we were trying to avoid via the use of hub events. 

I've implemented your suggestion 2).